miio/gateway/sub_device/switch.py

```diff
@@ -0,0 +1,245 @@
+"""Xiaomi Zigbee switches."""
+
+from enum import IntEnum
+
+import attr
+import click
+
+from ...click_common import command
+from .sub_device import SubDevice
+
+
+class OneChannelSwitch(SubDevice):
```

Comment:
@rytilahti all the switches have shared base classes, the structure is:
OneChannelSwitch = Main base class
TwoChannelSwitch inherits from OneChannelSwitch and is also a base class
OneChannelSwitchLoad inherits from OneChannelSwitch and is also a base class
TwoChannelSwitchLoad inherits from OneChannelSwitch and is also a base class
ThreeChannelSwitchLoad inherits from OneChannelSwitch and is also a base class

The rest of the classes in this file represent individual physicall devices and inherit from one of the base classes above.

Reply:
`switches` are devices that can actually physically turn something on/off, they can supply power to an appliance such as a lamp.
`remote_switches` are just battery powered things you can stick anywhere on a wall and can only give an signal to the gateway that can trigger an automation to do someting (like turn on/off a smart light), they are supposed to be mounted on a wall.
`buttons` are basically the same thing as `remote_switches` but they are small hand held battery powered devices.

I think we could put `buttons` and `remote_switches` together since they might have a simular protecoll.
`switches` defenitly have diffrent commands, so schould be seperated.

Reply:
I have put the `buttons` and `remote_switches` together in the remote_switches.py file